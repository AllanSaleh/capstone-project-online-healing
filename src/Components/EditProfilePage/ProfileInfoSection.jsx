import React from 'react';

import ProfileInputField from './ProfileInputField';
import BirthDateIcon from './Images/BirthDateIcon.svg';
import Plus from './Images/Plus.svg';

const ProfileInfoSection = ({ userProfile }) => (
  <form className="mb-8">
    <h1 className="text-3xl lg:text-title uppercase">Profile Info</h1>

    <ProfileInputField title="full name" type="text" id="full-name" value={userProfile?.fullname} />

    <div className="flex items-center my-6">
      <label
        htmlFor="education-level"
        className="text-sm md:text-lg lg:text-subtitle capitalize inline-block w-1/3"
      >
        education level
      </label>
      <div className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg px-6 py-2">
        <select
          id="education-level"
          name="education-level"
          className="w-full h-full outline-none py-1"
        >
          <option value="1">bachelor</option>
          <option value="2">bachelor</option>
          <option value="3">bachelor</option>
          <option value="4">bachelor</option>
        </select>
      </div>
    </div>
    <ProfileInputField title="hobbies" type="text" id="hobbies" value={userProfile?.hobbies} />

    <div className="flex items-center my-6">
      <label
        htmlFor="family-members"
        className="text-sm md:text-lg lg:text-subtitle capitalize inline-block w-1/3"
      >
        family members
      </label>
      <div className="w-2/12 lg:w-1/12 inline-flex items-center border border-black border-opacity-20 rounded-lg px-4">
        <input
          type="text"
          id="family-members"
          name="family-members"
          className="w-full h-full outline-none py-3"
          value={userProfile?.family_size}
        />
      </div>
      <span className="text-sm md:text-lg lg:text-subtitle capitalize mx-4">member(s)</span>
    </div>

    <div className="flex items-center my-6">
      <label
        htmlFor="gender"
        className="sm:text-sm md:text-lg lg:text-subtitle capitalize inline-block w-1/3"
      >
        gender
      </label>
      <div className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg px-6 py-2">
        <select id="gender" name="gender" className="w-full h-full outline-none py-1">
          <option value="1">Male</option>
          <option value="2">Female</option>
        </select>
      </div>
    </div>

    <ProfileInputField
      title="birth date"
      type="text"
      id="birth-date"
      icon={BirthDateIcon}
      value={userProfile?.birthdate}
    />
    <ProfileInputField title="email" type="email" id="email" value={userProfile?.email} />
    <ProfileInputField
      title="phone number"
      type="text"
      id="phone-number"
      value={userProfile?.phone_number}
    />
    <ProfileInputField title="upload ID" type="file" id="upload-id" icon={Plus} />
  </form>
);

export default ProfileInfoSection;
