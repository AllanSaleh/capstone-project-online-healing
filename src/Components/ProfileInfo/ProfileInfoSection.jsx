import React from 'react';
import ProfileInputField from './ProfileInputField';
import BirthDateIcon from '../../Icons/BirthDateIcon.svg';
import ArrowDown from '../../Icons/ArrowDown.svg';
import Plus from '../../Icons/Plus.svg';

const ProfileInfoSection = () => (
  <form className="my-4">
    <h1 className="text-title uppercase">Profile Info</h1>
    <ProfileInputField title="full name" type="text" id="full-name" />
    <ProfileInputField title="hobbies" type="text" id="hobbies" />
    <ProfileInputField title="birth date" type="text" id="birth-date" icon={BirthDateIcon} />
    <ProfileInputField title="email" type="email" id="email" />
    <ProfileInputField title="phone number" type="text" id="phone-number" />
    <ProfileInputField title="upload ID" type="file" id="upload-id" icon={Plus} />
  </form>
);

export default ProfileInfoSection;
