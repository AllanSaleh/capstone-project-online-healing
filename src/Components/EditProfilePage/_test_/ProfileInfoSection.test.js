import * as React from 'react';
import * as renderer from 'react-test-renderer';
import ProfileInfoSection from '../ProfileInfoSection';

it('Edit Info Section renders correctly', () => {
  const tree = renderer.create(<ProfileInfoSection />).toJSON();
  expect(tree).toMatchSnapshot();
});
