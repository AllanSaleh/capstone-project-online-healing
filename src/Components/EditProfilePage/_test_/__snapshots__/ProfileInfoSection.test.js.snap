// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Info Section renders correctly 1`] = `
<form
  className="mb-8"
>
  <h1
    className="text-title uppercase"
  >
    Profile Info
  </h1>
  <div
    className="flex items-center my-6"
  >
    <label
      className="text-subtitle capitalize inline-block w-1/3"
      htmlFor="full-name"
    >
      full name
    </label>
    <div
      className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4"
    >
      <input
        className="w-11/12 h-full outline-none py-3"
        id="full-name"
        name="full-name"
        type="text"
      />
    </div>
  </div>
  <div
    className="flex items-center my-6"
  >
    <label
      className="text-subtitle capitalize inline-block w-1/3"
      htmlFor="education-level"
    >
      education level
    </label>
    <div
      className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg px-6 py-2"
    >
      <select
        className="w-full h-full outline-none py-1"
        id="education-level"
        name="education-level"
      >
        <option
          value="1"
        >
          bachelor
        </option>
        <option
          value="2"
        >
          bachelor
        </option>
        <option
          value="3"
        >
          bachelor
        </option>
        <option
          value="4"
        >
          bachelor
        </option>
      </select>
    </div>
  </div>
  <div
    className="flex items-center my-6"
  >
    <label
      className="text-subtitle capitalize inline-block w-1/3"
      htmlFor="hobbies"
    >
      hobbies
    </label>
    <div
      className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4"
    >
      <input
        className="w-11/12 h-full outline-none py-3"
        id="hobbies"
        name="hobbies"
        type="text"
      />
    </div>
  </div>
  <div
    className="flex items-center my-6"
  >
    <label
      className="text-subtitle capitalize inline-block w-1/3"
      htmlFor="family-members"
    >
      family members
    </label>
    <div
      className="w-1/12 inline-flex items-center border border-black border-opacity-20 rounded-lg px-4"
    >
      <input
        className="w-full h-full outline-none py-3"
        id="family-members"
        name="family-members"
        type="text"
      />
    </div>
    <span
      className="text-subtitle capitalize mx-4"
    >
      member(s)
    </span>
  </div>
  <div
    className="flex items-center my-6"
  >
    <label
      className="text-subtitle capitalize inline-block w-1/3"
      htmlFor="gender"
    >
      gender
    </label>
    <div
      className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg px-6 py-2"
    >
      <select
        className="w-full h-full outline-none py-1"
        id="gender"
        name="gender"
      >
        <option
          value="1"
        >
          Male
        </option>
        <option
          value="2"
        >
          Female
        </option>
      </select>
    </div>
  </div>
  <div
    className="flex items-center my-6"
  >
    <label
      className="text-subtitle capitalize inline-block w-1/3"
      htmlFor="birth-date"
    >
      birth date
    </label>
    <div
      className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4"
    >
      <input
        className="w-11/12 h-full outline-none py-3"
        id="birth-date"
        name="birth-date"
        type="text"
      />
      <button
        type="button"
      >
        <img
          alt="profileIcon"
          src="BirthDateIcon.svg"
        />
      </button>
    </div>
  </div>
  <div
    className="flex items-center my-6"
  >
    <label
      className="text-subtitle capitalize inline-block w-1/3"
      htmlFor="email"
    >
      email
    </label>
    <div
      className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4"
    >
      <input
        className="w-11/12 h-full outline-none py-3"
        id="email"
        name="email"
        type="email"
      />
    </div>
  </div>
  <div
    className="flex items-center my-6"
  >
    <label
      className="text-subtitle capitalize inline-block w-1/3"
      htmlFor="phone-number"
    >
      phone number
    </label>
    <div
      className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4"
    >
      <input
        className="w-11/12 h-full outline-none py-3"
        id="phone-number"
        name="phone-number"
        type="text"
      />
    </div>
  </div>
  <div
    className="flex items-center my-6"
  >
    <label
      className="text-subtitle capitalize inline-block w-1/3"
      htmlFor="upload-id"
    >
      upload ID
    </label>
    <div
      className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4"
    >
      <input
        className="w-11/12 h-full outline-none py-3"
        id="upload-id"
        name="upload-id"
        type="file"
      />
      <button
        type="button"
      >
        <img
          alt="profileIcon"
          src="Plus.svg"
        />
      </button>
    </div>
  </div>
</form>
`;
